### PR TITLE
Cult shoes

### PR DESCRIPTION
**Pull request by MarsM0nd**
Gives cultists their shoes on usage of the armor talisman, updates the text in the tome to reflect that. Restored the behaviour that items you can't equiped won't get summoned, as described by the tome. This also fixes a possible exploit of unlimited storage space for cultists since `equip_or_collect()` did put the summoned backpack into your equiped backpack.
:cl:
add: Cultists get shoes on usage of the armor talisman.
add: Cultists only get equipment if they have the slot for it free, weapons are always spawned.
tweak: The tome does a more fitting description of the talisman of armor.
/:cl:

### CODE REVIEW

File: code/game/gamemodes/cult/talisman.dm [hunk 1]
```diff
@@ -272,6 +272,7 @@
 						 "<span class='cultitalic'>You speak the words of the talisman, arming yourself!</span>")
 	H.equip_or_collect(new /obj/item/clothing/suit/hooded/cultrobes/alt(user), slot_wear_suit)
 	H.equip_or_collect(new /obj/item/weapon/storage/backpack/cultpack(user), slot_back)
+	H.equip_or_collect(new /obj/item/clothing/shoes/cult(user), slot_back)
```
Comment (Fox-McCloud) [inline code/game/gamemodes/cult/talisman.dm]: I detect copy pasta.

shoes go on your feet, not your back.

**Comment by Necaladun** [comment]
Shoes are important. Style is important.

:+1:

File: code/game/gamemodes/cult/talisman.dm [hunk 2]
```diff
@@ -270,8 +270,13 @@
 	var/mob/living/carbon/human/H = user
 	user.visible_message("<span class='warning'>Otherworldly armor suddenly appears on [user]!</span>", \
 						 "<span class='cultitalic'>You speak the words of the talisman, arming yourself!</span>")
-	H.equip_or_collect(new /obj/item/clothing/suit/hooded/cultrobes/alt(user), slot_wear_suit)
-	H.equip_or_collect(new /obj/item/weapon/storage/backpack/cultpack(user), slot_back)
+	if(!H.wear_suit) //Only get the equipment if the slot is free.
```
Comment (Fox-McCloud) [inline code/game/gamemodes/cult/talisman.dm]: These are all kinda unnecessary; `equip_or_collect` ensures that if the slot is filled, it'll try to put it somewhere else.

File: code/game/gamemodes/cult/talisman.dm [hunk 3]
```diff
@@ -270,8 +270,13 @@
 	var/mob/living/carbon/human/H = user
 	user.visible_message("<span class='warning'>Otherworldly armor suddenly appears on [user]!</span>", \
 						 "<span class='cultitalic'>You speak the words of the talisman, arming yourself!</span>")
-	H.equip_or_collect(new /obj/item/clothing/suit/hooded/cultrobes/alt(user), slot_wear_suit)
-	H.equip_or_collect(new /obj/item/weapon/storage/backpack/cultpack(user), slot_back)
+	if(!H.wear_suit) //Only get the equipment if the slot is free.
+		H.equip_or_collect(new /obj/item/clothing/suit/hooded/cultrobes/alt(user), slot_wear_suit)
+	if(!H.back)
```
Comment (Fox-McCloud) [inline code/game/gamemodes/cult/talisman.dm]: Same here.

File: code/game/gamemodes/cult/talisman.dm [hunk 4]
```diff
@@ -270,8 +270,13 @@
...
 	user.visible_message("<span class='warning'>Otherworldly armor suddenly appears on [user]!</span>", \
 						 "<span class='cultitalic'>You speak the words of the talisman, arming yourself!</span>")
-	H.equip_or_collect(new /obj/item/clothing/suit/hooded/cultrobes/alt(user), slot_wear_suit)
-	H.equip_or_collect(new /obj/item/weapon/storage/backpack/cultpack(user), slot_back)
+	if(!H.wear_suit) //Only get the equipment if the slot is free.
+		H.equip_or_collect(new /obj/item/clothing/suit/hooded/cultrobes/alt(user), slot_wear_suit)
+	if(!H.back)
+		H.equip_or_collect(new /obj/item/weapon/storage/backpack/cultpack(user), slot_back)
+	if(!H.shoes)
+		if(H.has_organ_for_slot(slot_shoes)) //Might be empty for a reason.
```
Comment (Fox-McCloud) [inline code/game/gamemodes/cult/talisman.dm]: You'd still want the shoes if your foot got blown off; you can always get the foot re-attached later).

**Comment by MarsM0nd** [comment]
The check if the slot is free and for the shoes to be possible to be equiped is intentional, since in the Tome, there is this line: "Any items that cannot be equipped will not be summoned."
That could be changed, but there is another issue i had with the ``equip_or_collect()`` with backpacks, namely that the backpack does get put into your normal backpack, which is above the size limit to put items in, and it goes above the total limit in general. Exploitable by using the talisman multiple times, and you got a backpack filled with backpacks, and more storage space than you could ever want.
I agree that the second shoe check might be left out for potential re-attachment, but it does make it a bit more consistent in my eyes, and i find it relativly unlikely for someone to get surgery after they went loud by arming themselfs.

**Comment by tigercat2000** [comment]
@MarsM0nd use equip_to_slot_or_del if that's the behaviour you want, don't add pointless conditionals

**Comment by MarsM0nd** [comment]
Thanks for the pointer, didn't find that in either inventory.dm from mobs in general or humans. Only thing that might be nice is that the talisman gets used up before the items get summoned, so both of the weapons land in your hands if you have them both free, but i guess that would need some snowflaking, as the talisman needs to do it's work before it's known it worked, and can be deleted.

**Pull request merged**